lib/aws/json.ex

```diff
@@ -0,0 +1,23 @@
+defmodule AWS.JSON do
+  @moduledoc """
+  Specifies the behaviour of a JSON parser.
+
+  You can switch the default JSON parser which uses Poison underneath
+  by defining a different implementation by setting the `:json_module`
+  configuration in AWS.Client:
+
+     client = %AWS.Client{json_module: MyCustomJSONParser}
+     Aws.SNS.publish(client, %{})
+
+  """
+
+  @doc """
+  Encodes a map into JSON string. Raises in case of errors.
+  """
+  @callback encode!(input :: map(), options :: keyword()) :: String.t()
```

Comment:
I think we can call `encode_to_iodata!` as that returns iodata and is more efficient.

Reply:
Only for JSON encode! or should we do that for decode too? XML module as well?

Reply:
For both, i think returning iodata is the preferred format, so we should promote it. even though the built-in XML one likely won' t use it. :)